src/adera/Machines/RCSController.h

```diff
@@ -80,62 +83,29 @@ class SysMachineRCSController
      * @param cmdRot    [in] Commanded axis of rotation
      */
     static float thruster_influence(
-        Magnum::Vector3 posOset, Magnum::Vector3 direction,
-        Magnum::Vector3 cmdTransl, Magnum::Vector3 cmdRot);
+        osp::Vector3 posOset, osp::Vector3 direction,
+        osp::Vector3 cmdTransl, osp::Vector3 cmdRot);
 
 }; // SysMachineRCSController
 
 //-----------------------------------------------------------------------------
 
-class MachineRCSController : public osp::active::Machine
+class MachineRCSController
 {
     friend SysMachineRCSController;
 
+    using Percent = osp::active::wiretype::Percent;
+    using AttitudeControl = osp::active::wiretype::AttitudeControl;
+
 public:
 
     static inline std::string smc_mach_name = "RCSController";
 
-    MachineRCSController();
-    MachineRCSController(MachineRCSController&& move) noexcept;
-    MachineRCSController& operator=(MachineRCSController&& move) noexcept;
-    ~MachineRCSController() = default;
+    static constexpr osp::portindex_t<AttitudeControl> smc_wiCommandOrient{0};
 
-    void propagate_output(osp::active::WireOutput *output) override;
+    static constexpr osp::portindex_t<Percent> m_woThrottle{0};
 
-    osp::active::WireInput* request_input(osp::WireInPort port) override;
-    osp::active::WireOutput* request_output(osp::WireOutPort port) override;
-
-    std::vector<osp::active::WireInput*> existing_inputs() override;
-    std::vector<osp::active::WireOutput*> existing_outputs() override;
-
-private:
-    osp::active::WireInput m_wiCommandOrient{this, "Orient"};
-    osp::active::WireOutput m_woThrottle{this, "Throttle"};
-
-    osp::active::ActiveEnt m_rigidBody{entt::null};
```

Comment:
I find it's quite beautiful how the RCS controller had been made stateless, with all common properties separated into their own components. Like a logic gate, it's only purpose is to read a value from its input, and write a value to its output.